packages/langium/src/grammar/lsp/langium-grammar-reference-finder.ts

```diff
@@ -0,0 +1,212 @@
+/******************************************************************************
```

Comment:
You put a lot of effort in here: about 200 LOC. Wow.

As I mentioned [in Slack (internal)](https://typefox.slack.com/archives/C01M3T0BQ7L/p1654760192140229), we should think about ways of minimizing the duplicated code. Because that is a kind of risk. If you add another feature XY where type names are also part of, you are likely in a position where you could forget 1 of 3-4 places.

I think we can make this code much shorter (later), like
```typescript
graph.nodes.getByName(typeName) //every type has a node
           .getIncomingEdges()  //with a bunch of incoming or defining edges
           .filter(e => e.type == 'typedef') //filter out the right edges, if needed
           .flatMap(e => e.astNodes) //get the ast nodes that define the type
```

This is just an example.

Reply:
Not the right moment, but I will add an issue...